stack-macos.yaml

```diff
@@ -4,14 +4,14 @@
 # license that can be found in the LICENSE file or at
 # https://developers.google.com/open-source/licenses/bsd
 
-resolver: lts-16.26
+resolver: lts-17.5
 
 packages:
 - .
 
 extra-deps:
-  - github: apaszke/llvm-hs
-    commit: a9a74be1a7c15f3d21b2fffd35a425002ae7736f
+  - github: junjihashimoto/llvm-hs
+    commit: 90f47b9881e6c4fc4d1870e44b8c1075d8fcdc23
```

Comment:
Why this change?

Reply:
This dump is to fix [build errors](https://github.com/google-research/dex-lang/runs/2069515204
) for llvm-hs on macos.

There are two problems and it seems that the c++ default version and its settings are not working.

First problem is that the default version of c++ to build llvm-hs is different for linux and macos.
linux uses gcc, and macos uses clang on xcode.

gcc uses c++-14 by default.
https://gcc.gnu.org/projects/cxx-status.html
xcodes may use c++-98 by default.

Second problem is that ghc-8.8 had a bug in handling c++ options.
https://gitlab.haskell.org/ghc/ghc/-/issues/16477

The bump of llvm-hs seems to have solved these problems, because llvm-hs can be compiled.
Maybe the hash of the source I used is not good. The build of the dex package fails.
![image](https://user-images.githubusercontent.com/2469618/110710735-bf5d9b00-8241-11eb-920d-7cc286cf43f5.png)